zipkin-autoconfigure/metrics-prometheus/src/main/java/zipkin2/autoconfigure/prometheus/ZipkinPrometheusMetricsAutoConfiguration.java

```diff
@@ -65,12 +66,13 @@
   }
 
   // We need to make sure not-found requests are still handled by a service to be decorated for
-  // adding metrics. We add a lowest-precedence path mapping so anything not mapped by another
+  // adding metrics. We add a lower precedence path mapping so anything not mapped by another
   // service is handled by this.
   @Bean
-  @Order
+  @Order(1)
   ArmeriaServerConfigurator notFoundMetricCollector() {
-    return sb -> sb.serviceUnder("/", (ctx, req) -> HttpResponse.of(HttpStatus.NOT_FOUND));
+    // Use glob instead of catch-all to avoid adding it to the trie router.
+    return sb -> sb.service(PathMapping.ofGlob("/**"), (ctx, req) -> HttpResponse.of(HttpStatus.NOT_FOUND));
```

Comment:
@adriancole I fixed here to pass the test https://github.com/openzipkin/zipkin/blob/master/zipkin-server/src/test/java/zipkin2/server/internal/ITZipkinServerHttpCollectorDisabled.java#L49-L56. From the previous version, `405 method not allowed` returned because `/api/v1/spans` path exists but `POST` method is not allowed due to `zipkin.collector.http.enabled=false`. In this case `405 method not allowed` seems more reasonable, so would it better to fix the test code? Please let me know what you think! :-)